### ISSUE TITLE
Add chicken-bit support to BOOM

### ISSUE THREAD

**Issue by ccelio** [riscv-boom/riscv-boom#36]
**Summary:**
A nice-to-have feature is the ability to enable and disable hardware features from the software.

**Examples:**
* Enable TSO memory consistency
* Enable watchdog timer
* Change predictor behaviors

**High-level thoughts:**
* Do **not** use non-standard Control/Status Registers (CSRs). The address space will get too congested, not indexable, messy to assemble for, etc.
* Instead provide a tile control port MMIO device. Rocket-chip currently uses one to report ECC and bus errors.
* It would be nice to provide a systematic interface so that BOOM control signals can register themselves as part of the chicken-bit system.

**Comment by ccelio** [comment]
The new chisel3.util.experimental.BoringUtils may help here.

**Comment by akshataj96** [comment]
Hi, Is this still available to pick up?

**Comment by jerryz123** [comment]
Yes, we would be happy to accept PRs adding new chicken bits to BOOM.

**Comment by akshataj96** [comment]
Okay great! I will start working on it and revert if I have any questions.

**Comment by 124107157-KV** [comment]
Hi, I’m interested in working on this issue as my first contribution to BOOM.

I have experience with RISC-V RTL, processor control paths, runtime-selectable hardware modes, and verification, although I’m still building familiarity with the BOOM/Chisel codebase.

Since this issue was opened some time ago, could you please confirm whether the proposed tile-control MMIO approach is still preferred? I would like to begin with a small, review-able contribution, possibly adding one chicken bit for an existing BOOM feature, together with documentation and a directed test.

Is there a particular feature or predictor behavior that would be suitable as the first target?
